sympy/core/expr.py

```diff
@@ -843,6 +843,12 @@ def equals(self, other, failing_expression=False):
         # calculated twice as the same value.
         if constant not in (True, None) and constant != 0:
             return False
+        
+        from sympy import radsimp
+        self = radsimp(simplify(self))
+        other = radsimp(other)
+        if self == other:
+            return True
 
```

Comment:
I don't think that this really addresses the underlying problem. I made some suggestions here: https://github.com/sympy/sympy/issues/25915#issuecomment-1824604943